eng/pipelines/release-go-images-pipeline.yml

```diff
@@ -138,14 +160,26 @@ extends:
                       reason: queued build
 
                 # Now we have poll2MicrosoftGoImagesBuildID
-                - script: |
-                    releasego wait-build \
-                      -id '$(poll2MicrosoftGoImagesBuildID)' \
-                      -org 'https://dev.azure.com/dnceng/' \
-                      -proj 'internal' \
-                      -azdopat '$(System.AccessToken)'
-                  displayName: ⌚ Wait for go-images build
-                  timeoutInMinutes: 120
+                - ${{ if eq(parameters.runGoImagesBuild, true) }}:
```

Comment:
Thinking again about this, this shouldn't be necessary. If you run a build with `poll2MicrosoftGoImagesBuildID` filled in to point at an already-completed go-images build, then the polling step will instantly pass. This is what I filled in for https://dev.azure.com/dnceng/internal/_build/results?buildId=2466716&view=logs&j=19992227-62fb-5b50-4e29-3b72bd33eea1. (That link should take you directly to a log where you can expand a `Parent pipeline used these runtime parameters` line to see what I filled in.) If you go to that build and hit "Run new", AzDO copies these values into the dialog box, which is helpful to make it less tedious to fill in. This workflow is what I was trying to suggest in the team sync chat.

For convenience's sake (not needing to grab an ID and fill in that box), I don't disagree with this feature, but it needs a fix. Currently this build triggers a go-images build even if `runGoImagesBuild` is `false`, because this `if` only applies to `releasego wait-build`, not `releasego build-pipeline`. The condition needs to be copied to apply to the whole other block--`releasego get-images-commit`,  `releasego wait-azdo-commit`, `releasego build-pipeline`, and the report with `reason: queued build`.

Spinning the go-images build without a need is just a waste of resources and shouldn't cause any functional problems, but it should be avoided. 😄